run_project_tests.py

```diff
@@ -196,6 +196,16 @@ def platform_fix_name(fname):
         if mesonlib.is_windows() or mesonlib.is_cygwin():
             return fname + '.exe'
 
+    if fname.startswith('?msvc:'):
+        fname = fname[6:]
+        if compiler != 'cl':
+            return None
+
+    if fname.startswith('?gcc:'):
+        fname = fname[5:]
+        if compiler == 'cl':
+            return None
+
```

Comment:
While you're at it, can you also consolidate the other windows tests that are divided into separate mingw and msvc tests?